core/src/syscall/precompiles/blake2b/compress/mix.rs

```diff
@@ -0,0 +1,281 @@
+use core::borrow::Borrow;
+use core::borrow::BorrowMut;
+use p3_field::Field;
+
+use sp1_derive::AlignedBorrow;
+use std::mem::size_of;
+
+use crate::air::WORD_SIZE;
+use crate::air::{SP1AirBuilder, Word};
+use crate::operations::{AddOperationU64, FixedRotateRightOperationU64, XorOperation};
+use crate::runtime::ExecutionRecord;
+
+use super::mix;
+use super::{MIX_INPUT_SIZE, STATE_ELE_PER_CALL};
+/// A set of columns needed to compute the `mix` of the input state.
+///  ``` ignore
+/// fn mix(state: &mut BlockWords, a: u64, b: u64, c: u64, d: u64, x: u64, y: u64) {
+///     state[a] = state[a].wrapping_add(state[b]).wrapping_add(x);
+///     state[d] = (state[d] ^ state[a]).rotate_right(32);
+///     state[c] = state[c].wrapping_add(state[d]);
+///     state[b] = (state[b] ^ state[c]).rotate_right(24);
+///     state[a] = state[a].wrapping_add(state[b]).wrapping_add(y);
+///     state[d] = (state[d] ^ state[a]).rotate_right(16);
+///     state[c] = state[c].wrapping_add(state[d]);
+///     state[b] = (state[b] ^ state[c]).rotate_right(63);
+/// }
+///  ```
+#[derive(AlignedBorrow, Default, Debug, Clone, Copy)]
+#[repr(C)]
+pub struct MixOperation<T> {
+    pub a_plus_b: AddOperationU64<T>,
+    pub a_plus_b_plus_x: AddOperationU64<T>,
+    pub d_xor_a_lo: XorOperation<T>,
```

Comment:
hm, perhaps we should have a `Xor64Operation` that is very simple and just wraps 2 `XorOperation`'s, what do you think?

Reply:
yeah, it makes sense! I've introduced it in latest commit https://github.com/succinctlabs/sp1/commit/dae55542cd38371935a43e3c8421badded586abb